src/Storefront/Resources/app/storefront/src/plugin/address-search/address-search.plugin.js

```diff
@@ -0,0 +1,42 @@
+import Plugin from 'src/plugin-system/plugin.class';
+
+/**
+ * @package checkout
+ */
+export default class AddressSearchPlugin extends Plugin {
+
+    static options = {
+        searchQuerySelector: '.address-manager-modal-list-search',
+        searchContentQuerySelector: '.address-manager-modal-list-wrapper',
+        searchItemQuerySelector: '.address-manager-modal-select-address',
+    };
+
+    init() {
+        this._registerEvents();
+    }
+
+    /**
+     * registers all needed event listeners
+     *
+     * @private
+     */
+    _registerEvents() {
+        this.el.querySelector(this.options.searchQuerySelector)?.addEventListener('input', this._onSearch.bind(this));
+    }
+
+    _onSearch(event) {
+        const searchQuery = event.target.value.toLowerCase();
+
+        const items = this.el.querySelector(this.options.searchContentQuerySelector).children;
+
+        Array.from(items).forEach(item => {
+            const text = item.querySelector(this.options.searchItemQuerySelector).textContent.toLowerCase();
+
+            if (text.includes(searchQuery)) {
+                item.style.display = '';
+            } else {
+                item.style.display = 'none';
+            }
```

Comment:
This might be a cleaner solution and more in line with other plugins:
```suggestion
            if (text.includes(searchQuery)) {
                item.classList.remove('d-none')
            } else {
                item.classList.add('d-none')
            }
```